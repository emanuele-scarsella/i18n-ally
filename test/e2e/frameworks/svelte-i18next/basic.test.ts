import { window } from 'vscode'
import { getExt, openFile, Global, is, not, setupTest, expect, timeout, KeyDetector } from '../../ctx'

setupTest('Svelte with i18next', () => {
  it('opens entry file', async() => {
    await openFile('package.json')
  })

  it('is active', () => {
    const ext = getExt()
    is(ext?.isActive, true)
  })

  it('enables correct frameworks', async() => {
    not(Global, undefined)
    is(Global.enabled, true)
    is(Global.enabledFrameworks.length, 1)
    is(Global.enabledFrameworks[0].id, 'svelte-i18next')
  })

  it('get correct coverage report', async() => {
    await timeout(500)
    not(Global, undefined)
    not(Global.loader, undefined)
    const coverage_en = Global.loader.getCoverage('en')
    expect(coverage_en).to.matchSnapshot()
    const coverage_de = Global.loader.getCoverage('de')
    expect(coverage_de).to.matchSnapshot()
  })

  it('get keys', async() => {
    await openFile('src/App.svelte')
    await timeout(500)
    const keys = KeyDetector.getKeys(window.activeTextEditor!.document)
    expect(keys).to.matchSnapshot()
  })
})
