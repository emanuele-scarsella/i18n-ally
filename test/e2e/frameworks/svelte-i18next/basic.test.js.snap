// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svelte with i18next get correct coverage report 1`] = `
Object {
  "allKeys": Array [
    "pages.home.title",
    "translation.foo.bar",
    "translation.title",
  ],
  "emptyKeys": Array [],
  "locale": "en",
  "missing": 0,
  "missingKeys": Array [],
  "total": 3,
  "translated": 3,
  "translatedKeys": Array [
    "pages.home.title",
    "translation.foo.bar",
    "translation.title",
  ],
}
`;

exports[`Svelte with i18next get correct coverage report 2`] = `
Object {
  "allKeys": Array [
    "pages.home.title",
    "translation.foo.bar",
    "translation.title",
  ],
  "emptyKeys": Array [],
  "locale": "de",
  "missing": 0,
  "missingKeys": Array [],
  "total": 3,
  "translated": 3,
  "translatedKeys": Array [
    "pages.home.title",
    "translation.foo.bar",
    "translation.title",
  ],
}
`;

exports[`Svelte with i18next get keys 1`] = `
Array [
  Object {
    "end": 93,
    "key": "title",
    "quoted": true,
    "start": 88,
  },
  Object {
    "end": 201,
    "key": "title",
    "quoted": true,
    "start": 196,
  },
  Object {
    "end": 463,
    "key": "pages.home.title",
    "quoted": true,
    "start": 447,
  },
  Object {
    "end": 494,
    "key": "foo.bar",
    "quoted": true,
    "start": 487,
  },
]
`;
